mirascope/llm/_base_provider_converter.py

```diff
@@ -0,0 +1,50 @@
+from abc import ABC, abstractmethod
+from typing import Any, Callable
+
+from pydantic import BaseModel
+
+from mirascope.core import BaseMessageParam
+from mirascope.core.base import BaseCallResponse, BaseCallResponseChunk, BaseStream
+from mirascope.llm.call_response_chunk import FinishReason
+
+
+class Usage(BaseModel):
+    completion_tokens: int
+    """Number of tokens in the generated completion."""
+
+    prompt_tokens: int
+    """Number of tokens in the prompt."""
+
+    total_tokens: int
+    """Total number of tokens used in the request (prompt + completion)."""
+
+
+class BaseProviderConverter(ABC):
```

Comment:
I decided to aggregate the specific logic for each provider into this class.
What do you think?